Remove `prospector` from requirements

**Describe the bug**
The package `prospector` is no longer used and it is causing issues during `pip` installation.
Error:
```
 Collecting prospector
  Downloading prospector-1.8.2-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.3/85.3 kB 31.4 MB/s eta 0:00:00
  Downloading prospector-1.8.1-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.3/85.3 kB 8.1 MB/s eta 0:00:00
  Downloading prospector-1.8.0-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.4/85.4 kB 24.2 MB/s eta 0:00:00
  Downloading prospector-1.7.7-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.9/85.9 kB 31.2 MB/s eta 0:00:00
  Downloading prospector-1.7.6-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.7/85.7 kB 31.8 MB/s eta 0:00:00
  Downloading prospector-1.7.5-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.3/85.3 kB 26.9 MB/s eta 0:00:00
  Downloading prospector-1.7.4-py3-none-any.whl (84 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.5/84.5 kB 31.8 MB/s eta 0:00:00
  Downloading prospector-1.7.1-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.3/83.3 kB 31.7 MB/s eta 0:00:00
  Downloading prospector-1.7.0-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.4/83.4 kB 31.9 MB/s eta 0:00:00
  Downloading prospector-1.6.0-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.5/83.5 kB 30.5 MB/s eta 0:00:00
  Downloading prospector-1.5.3.1-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.5/83.5 kB 31.2 MB/s eta 0:00:00
  Downloading prospector-1.5.3-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.5/83.5 kB 29.9 MB/s eta 0:00:00
  Downloading prospector-1.5.2-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.6/83.6 kB 31.4 MB/s eta 0:00:00
  Downloading prospector-1.5.1-py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.6/83.6 kB 34.0 MB/s eta 0:00:00
   Downloading prospector-1.5.0.1-py3-none-any.whl (84 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.5/84.5 kB 26.0 MB/s eta 0:00:00
Collecting setoptconf<0.3.0,>=0.2.0
  Downloading setoptconf-0.2.0.tar.gz (13 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in setoptconf setup command: use_2to3 is invalid.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: Process completed with exit code 1.
```

**To Reproduce**
It happens randomly at CI, example: https://github.com/bihealth/cubi-tk/actions/runs/3676861244/jobs/6218051102

**Expected behavior**
PIP installation should go through without incidents.

**Additional context**
Related PR: #87

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ pytest:
 	pytest .
 
 .PHONY: lint-all
-lint-all: bandit pyflakes pep257 prospector
+lint-all: bandit pyflakes pep257
 
 .PHONY: bandit
 bandit:
@@ -31,7 +31,3 @@ pep257:
 .PHONY: flake8
 flake8:
 	flake8
-
-.PHONY: prospector
-prospector:
-	prospector

--- a/requirements/test.txt
+++ b/requirements/test.txt
@@ -38,7 +38,5 @@ mypy
 # More checking
 bandit
 pyflakes
-prospector
 
-# Pin version to avoid python 2 to 3 breaks (prospector)
-setuptools ==58.0.0
+setuptools ==65.6.3

